activerecord/test/cases/base_test.rb

```diff
@@ -1122,15 +1123,14 @@ def test_no_limit_offset
     end
   end
 
-  def test_find_last
-    last  = Developer.last
-    assert_equal last, Developer.all.merge!(:order => 'id desc').first
-  end
-
   def test_last
     assert_equal Developer.all.merge!(:order => 'id desc').first, Developer.last
   end
 
+  def test_first_and_last_are_equal_without_primary_key
+    assert_equal Season.first, Season.last
```

Comment:
This feels nonsensical to me. I think a better behavior would be to raise if `last` is called with no order and no primary key.